.github/workflows/e2e-bittensor-tests.yml

```
@@ -0,0 +1,84 @@
+name: E2E Bittensor Tests
+
+concurrency:
+  group: e2e-bittensor-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  ## Run automatically for all PRs against main, regardless of what the changes are
+  ## to be safe and so we can more easily force re-run the CI when github is being
+  ## weird by using a blank commit
+  push:
+    branches: [main, development, staging]
+
+  ##
+  # Run automatically for PRs against default/main branch if Rust files change
+  pull_request:
+    branches: [main, development, staging]
+
+  ## Allow running workflow manually from the Actions tab
+  workflow_dispatch:
+    inputs:
+      verbose:
+        description: "Output more information when triggered manually"
+        required: false
+        default: ""
+
+env:
+  CARGO_TERM_COLOR: always
+  VERBOSE: ${{ github.events.input.verbose }}
+
+jobs:
+  run:
+    runs-on: SubtensorCI
+    strategy:
+      matrix:
+        rust-branch:
+          - nightly-2024-03-05
+        rust-target:
+          - x86_64-unknown-linux-gnu
+          # - x86_64-apple-darwin
+        os:
+          - ubuntu-latest
+          # - macos-latest
+        include:
+          - os: ubuntu-latest
+          # - os: macos-latest
+    env:
+      RELEASE_NAME: development
+      RUSTV: ${{ matrix.rust-branch }}
+      RUST_BACKTRACE: full
+      RUST_BIN_DIR: target/${{ matrix.rust-target }}
+      TARGET: ${{ matrix.rust-target }}
+    steps:
+      - name: Check-out repository under $GITHUB_WORKSPACE
+        uses: actions/checkout@v2
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update &&
+          sudo apt-get install -y clang curl libssl-dev llvm libudev-dev protobuf-compiler
+
+      - name: Install Rust ${{ matrix.rust-branch }}
+        uses: actions-rs/toolchain@v1.0.6
+        with:
+          toolchain: ${{ matrix.rust-branch }}
+          components: rustfmt
+          profile: minimal
+
+      - name: Clone bittensor repo
+        run: git clone https://github.com/opentensor/bittensor.git
+
+      - name: Setup bittensor repo
+        working-directory: ${{ github.workspace }}/bittensor
+        run: |
+          git checkout tests/e2e-tests-staging
```

TODO before merge, once https://github.com/opentensor/bittensor/pull/1885 is in staging
```suggestion
          git checkout staging
```